go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -431,7 +432,11 @@ func (tm *TabletManager) demotePrimary(ctx context.Context, revertPartialFailure
 	if *setSuperReadOnly {
 		// Setting super_read_only also sets read_only
 		if err := tm.MysqlDaemon.SetSuperReadOnly(true); err != nil {
-			return nil, err
+			if strings.Contains(err.Error(), strconv.Itoa(mysql.ERUnknownSystemVariable)) {
+				log.Warningf("server does not know about super_read_only; maybe MariaDB? Continuing anyway.")
```

Comment:
I'd remove the maybe MariaDB part as the user could be using MySQL 5.6 or who knows what. MariaDB may eventually add the option too. 

Although I still think that we should end any official support for MariaDB as it's so different now and we don't test with it. That's another discussion though :-)